packages/@orbit/record-cache/src/live-query/live-query.ts

```diff
@@ -0,0 +1,187 @@
+import { Evented } from '@orbit/core';
+import {
+  QueryExpression,
+  FindRecord,
+  FindRecords,
+  FindRelatedRecord,
+  FindRelatedRecords,
+  equalRecordIdentities,
+  Query,
+  Schema,
+  RecordOperation,
+  RecordException
+} from '@orbit/data';
+
+import { QueryResult } from '../query-result';
+import { recordOperationChange, RecordChange } from './utils';
+
+export interface LiveQuerySettings {
+  query: Query;
+}
+
+export class LiveQuery {
+  cache: Evented;
+  schema: Schema;
+  query: Query;
+
+  executeQuery(
+    onNext: (result: QueryResult) => void,
+    onError?: (error: RecordException) => void
+  ): void {
+    throw new TypeError('executeQuery: Not Implemented.');
+  }
+
+  on(
+    onNext: (result: QueryResult) => void,
+    onError?: (error: RecordException) => void
+  ): () => void {
+    const executeQuery = onceTick(() => this.executeQuery(onNext, onError));
+    const unsubscribePatch = this.cache.on(
+      'patch',
+      (operation: RecordOperation) => {
+        if (this.match(operation)) {
+          executeQuery();
+        }
+      }
+    );
+
+    const unsubscribeReset = this.cache.on('reset', () => {
+      executeQuery();
+    });
+
+    executeQuery();
```

Comment:
@tchak any reason to immediately execute the listener ?  Does it make sense to add an api for this similar to ember debounce method `debounce (target, method, args*, wait, immediate)`

Reply:
I made another change to the interface. So first execution needs to be explicit now.

```ts
const subscription = store.cache
  .liveQuery(q => q.findRecords('planet'))
  .subscribe(fn);

subscription.execute();
```

```ts
subscribe(): SyncLiveQuerySubscription;

interface SyncLiveQuerySubscription {
  unsubscribe(): void;
  execute(): QueryResult;
}
```

Reply:
I'm curious about the `store.cache` usage here, does this mean that liveQuery is only available on the local cache, not the actual store? And would this mean that a coordinator strategy that auto-pulls data from a remote would not be part of this liveQuery process?

Reply:
`liveQuery` as defined in this PR is a `cache` feature not a `source` feature. I would like to have a conversation about remote subscriptions, but it is a quite different topic. Regarding a setup with a remote source – if your remote source is syncing back to memory source 'liveQuery' will pick up changes.

Reply:
Ah, I see, thanks for explaining! Our team is currently looking at orbitjs as a backing store for our json:api connected react application. One thing I'd like to solve is auto-pull of missing data, together with a reactive connection to the query that triggered the pull. It seems like this is not quite there yet - We'll continue to watch this space :)

Reply:
so would it help you if in the `query`/`pull` flow there was a `subscribe`/`unsubscribe` event on the `source`? Or something similar. What is your server implementation like? Is your server able to push diff operations for a registered query?

Reply:
For example in one app I am trying to use something like this : https://gist.github.com/tchak/2a23843c35617a265e75c136afc99b77